### PR TITLE
Some more debug info

### PR DESCRIPTION
**Pull request by maxgalli**

### CODE REVIEW

File: optimization.py [hunk 1]
```diff
@@ -132,20 +132,19 @@ def merge_datasets(self):
...
 
-    def print_merged_graphs(self):
+    def get_pretty_printed_merged_graphs(self):
         def call_node_rec(nd):
             return PrintedNode(nd.__repr__())([call_node_rec(child) for child in nd.children])
         fancy_trees = [call_node_rec(graph) for graph in self.graphs]
-        logger.debug('Merged graphs:')
-        logger.debug('\n\n'.join([
-            drawTree2(False)(False)(t) for t in fancy_trees]))
+        return '\n\n'.join([drawTree2(False)(False)(t) for t in fancy_trees])
```
Comment (stwunsch) [inline optimization.py:147]: What are the dependencies here? Sry haven't followed the implementation here.

File: run.py [hunk 2]
```diff
@@ -51,7 +51,11 @@ def _run_multiprocess(self, graph):
             len(ptrs)))
         results = list()
         for ptr in ptrs:
+            start = time()
             th = ptr.GetValue()
+            end = time()
+            logger.debug('%%%%%%%%%% Event loop for graph {} run in {} seconds'.format(
```
Comment (stwunsch) [inline run.py]: Could you remove the `%%%%%%%`? We have the `DEBUG` flag for this.

Comment (stwunsch) [reply run.py]: Ah, and we don't want the measurement there. We need it from the start of the function to the end. So to measure the "inclusive selftime" of the function.

File: optimization.py (hunk 1 above)
Comment (maxgalli) [reply optimization.py:147]: no dependencies, I included the thing in the printing.py module

Comment (stwunsch) [reply optimization.py:147]: Alright, thanks!

**Comment by stwunsch** [comment]
Perfect!

**Pull request merged**